### ISSUE TITLE
Mongoid 5 update broke migrations 

### ISSUE THREAD

**Issue by griff** [errbit/errbit#921]
The pull request that upgraded to mongoid 5 also removed the ```mongoid_rails_migrations``` gem so that ```rake db:migration``` now does nothing.

I tried to manually add ```mongoid_rails_migrations``` back but it seems that mongoid 5 also broke the actual migrations and I get this output:

```
Overwriting existing field _id in class App.
Creating scope :page. Overwriting existing method NotificationServices::CampfireService.page.
Creating scope :page. Overwriting existing method NotificationServices::FlowdockService.page.
Creating scope :page. Overwriting existing method NotificationServices::GtalkService.page.
Creating scope :page. Overwriting existing method NotificationServices::HipchatService.page.
Creating scope :page. Overwriting existing method NotificationServices::HoiioService.page.
Creating scope :page. Overwriting existing method NotificationServices::HubotService.page.
Creating scope :page. Overwriting existing method NotificationServices::PushoverService.page.
Creating scope :page. Overwriting existing method NotificationServices::SlackService.page.
Creating scope :page. Overwriting existing method NotificationServices::WebhookService.page.
Migrating to MoveNoticesToSeparateCollection (20110422152027)
==  MoveNoticesToSeparateCollection: migrating ================================
NOTE: Mongoid.default_session is deprecated; use default_client instead. It will be removed on or after 2015-12-01.
Mongoid.default_session called from /opt/errbit/app/vendor/bundle/ruby/2.1.0/gems/mongoid_rails_migrations-1.0.1/lib/mongoid_rails_migrations/active_record_ext/migrations.rb:147.
rake aborted!
StandardError: An error has occurred, 20110422152027 and all later migrations canceled:

wrong number of arguments (1 for 0)/opt/errbit/app/db/migrate/20110422152027_move_notices_to_separate_collection.rb:6:in `select'
/opt/errbit/app/db/migrate/20110422152027_move_notices_to_separate_collection.rb:6:in `up'
/opt/errbit/app/vendor/bundle/ruby/2.1.0/gems/mongoid_rails_migrations-1.0.1/lib/mongoid_rails_migrations/active_record_ext/migrations.rb:85:in `block in migrate'
/opt/errbit/app/vendor/bundle/ruby/2.1.0/gems/mongoid_rails_migrations-1.0.1/lib/mongoid_rails_migrations/active_record_ext/migrations.rb:85:in `migrate'
/opt/errbit/app/vendor/bundle/ruby/2.1.0/gems/mongoid_rails_migrations-1.0.1/lib/mongoid_rails_migrations/active_record_ext/migrations.rb:169:in `migrate'
/opt/errbit/app/vendor/bundle/ruby/2.1.0/gems/mongoid_rails_migrations-1.0.1/lib/mongoid_rails_migrations/active_record_ext/migrations.rb:319:in `block in migrate'
/opt/errbit/app/vendor/bundle/ruby/2.1.0/gems/mongoid_rails_migrations-1.0.1/lib/mongoid_rails_migrations/active_record_ext/migrations.rb:297:in `each'
/opt/errbit/app/vendor/bundle/ruby/2.1.0/gems/mongoid_rails_migrations-1.0.1/lib/mongoid_rails_migrations/active_record_ext/migrations.rb:297:in `migrate'
/opt/errbit/app/vendor/bundle/ruby/2.1.0/gems/mongoid_rails_migrations-1.0.1/lib/mongoid_rails_migrations/active_record_ext/migrations.rb:203:in `up'
/opt/errbit/app/vendor/bundle/ruby/2.1.0/gems/mongoid_rails_migrations-1.0.1/lib/mongoid_rails_migrations/active_record_ext/migrations.rb:188:in `migrate'
/opt/errbit/app/vendor/bundle/ruby/2.1.0/gems/mongoid_rails_migrations-1.0.1/lib/mongoid_rails_migrations/mongoid_ext/railties/database.rake:42:in `block (2 levels) in <top (required)>'
ArgumentError: wrong number of arguments (1 for 0)
/opt/errbit/app/db/migrate/20110422152027_move_notices_to_separate_collection.rb:6:in `select'
/opt/errbit/app/db/migrate/20110422152027_move_notices_to_separate_collection.rb:6:in `up'
/opt/errbit/app/vendor/bundle/ruby/2.1.0/gems/mongoid_rails_migrations-1.0.1/lib/mongoid_rails_migrations/active_record_ext/migrations.rb:85:in `block in migrate'
/opt/errbit/app/vendor/bundle/ruby/2.1.0/gems/mongoid_rails_migrations-1.0.1/lib/mongoid_rails_migrations/active_record_ext/migrations.rb:85:in `migrate'
/opt/errbit/app/vendor/bundle/ruby/2.1.0/gems/mongoid_rails_migrations-1.0.1/lib/mongoid_rails_migrations/active_record_ext/migrations.rb:169:in `migrate'
/opt/errbit/app/vendor/bundle/ruby/2.1.0/gems/mongoid_rails_migrations-1.0.1/lib/mongoid_rails_migrations/active_record_ext/migrations.rb:319:in `block in migrate'
/opt/errbit/app/vendor/bundle/ruby/2.1.0/gems/mongoid_rails_migrations-1.0.1/lib/mongoid_rails_migrations/active_record_ext/migrations.rb:297:in `each'
/opt/errbit/app/vendor/bundle/ruby/2.1.0/gems/mongoid_rails_migrations-1.0.1/lib/mongoid_rails_migrations/active_record_ext/migrations.rb:297:in `migrate'
/opt/errbit/app/vendor/bundle/ruby/2.1.0/gems/mongoid_rails_migrations-1.0.1/lib/mongoid_rails_migrations/active_record_ext/migrations.rb:203:in `up'
/opt/errbit/app/vendor/bundle/ruby/2.1.0/gems/mongoid_rails_migrations-1.0.1/lib/mongoid_rails_migrations/active_record_ext/migrations.rb:188:in `migrate'
/opt/errbit/app/vendor/bundle/ruby/2.1.0/gems/mongoid_rails_migrations-1.0.1/lib/mongoid_rails_migrations/mongoid_ext/railties/database.rake:42:in `block (2 levels) in <top (required)>'
Tasks: TOP => db:migrate
(See full trace by running task with --trace)
```

**Comment by stevecrozz** [comment]
Thanks for the quick find. It looks like we'll need to figure this one out to get migrations running again.

**Comment by stevecrozz** [comment]
This may require some more effort than I thought. mongoid_rails_migrations doesn't support the latest mongoid or the latest rails and the author doesn't seem interested in spending time on it. It'll be up to us to decide in what capacity we want to continue using this project.

As a short term measure, we can instruct users to upgrade first to v0.4.0, run the migrations and then upgrade from there, but that does leave us frozen at the current migration version. I'd appreciate any thoughts from the community on what you want to do here. The only options I see are 1) delete all the migrations and leave ourselves frozen until some future time when we need more migrations and have the time to figure it out, or 2) spend the effort now to update or rebuild something like mongoid_rails_migrations.

**Comment by stevecrozz** [comment]
@griff Here's what I'm planning. I'm going to remove all the migrations from the codebase except the most recent one because all the others are quite old and I think we can safely stop maintaining them. Then I'll add documentation to let users know that if they're upgrading from a version prior to v0.3.0, then they'll need to upgrade to v0.3.0, run migrations, and then continue on to the version they really want. New deployments won't run into these issues because they won't have any of these old migrations hanging around.

I just tested this and its working for me after adding back the mongoid_rails_migrations gem. I'll have a commit probably by tomorrow for you to try if you're so inclined.

**Comment by griff** [comment]
That would work for me. I temporarily worked around this issue by just using the commit just before the mongoid 5 merge.

**Comment by stevecrozz** [comment]
Great. This issue should be solved now as of #925
